### DISCUSSION TITLE
dockerd and lxc for Linux

### DISCUSSION THREAD

**Discussion by siadat** [Homebrew/discussions#4074] [Linux]
### Additional Info

- [ ] I am using Windows Subsystem for Linux

### Output of `brew config`

```shell
HOMEBREW_VERSION: 3.6.16-76-g819aa09
ORIGIN: https://github.com/Homebrew/brew
HEAD: 819aa09e63d1e654fa66287db5814a616f114ee1
Last commit: 12 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 903882f943df1a6857dfdf6aba55506d39b3bbe0
Core tap last commit: 4 hours ago
Core tap branch: master
HOMEBREW_PREFIX: /home/linuxbrew/.linuxbrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_EDITOR: nvim
HOMEBREW_MAKE_JOBS: 4
Homebrew Ruby: 2.6.8 => /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.8_1/bin/ruby
CPU: quad-core 64-bit skylake
Clang: N/A
Git: 2.39.0 => /home/linuxbrew/.linuxbrew/bin/git
Curl: 7.58.0 => /usr/bin/curl
Kernel: Linux 4.15.0-200-generic x86_64 GNU/Linux
OS: Ubuntu 18.04.6 LTS (bionic)
Host glibc: 2.27
/usr/bin/gcc: 7.5.0
/usr/bin/ruby: N/A
glibc: 2.35_1
gcc@11: N/A
gcc: 12.2.0
xorg: N/A
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

I am experimenting with using Homebrew as a package in an Ubuntu box.

I noticed that the `docker` formula installs [docker cli](https://github.com/Homebrew/homebrew-core/blob/eafdc2c7ae5690d2858e77b24b457b3a98c0bd1d/Formula/docker.rb#L4), which makes sense for a macOS system. However, I'd like to install `dockerd` as well.

I see there are some Linux-specific formulae (specifying `depends_on :linux`), such as `strace`. Would it make sense to create formulae for `dockerd` and `lxc` as well?

**Comment by siadat** [comment]
I decided to install docker engine via apt-get for now and let systemctl start/monitor it.

Would it be reasonable to say that it makes sense to install long-running services in this way, otherwise I would have to find a way to ensure dockerd is running?

**Reply by SMillerDev** [reply]
That completely depends what you want from it.

**Reply by siadat** [reply]
That makes sense, thanks!

**Comment by SMillerDev** [comment]
If you can get it to build `dockerd`, pull requests are welcome.

**Comment by AkihiroSuda** [comment]
`dockerd` is now available as the `docker-engine` formula:
- https://github.com/Homebrew/homebrew-core/pull/270750
